main.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  # my-each
+
+  def my_each
+    ind = 0
+
+    while ind != length
+      yield self[ind]
+      ind += 1
+    end
+
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    index = 0
+    my_each do |el|
+      yield(el, index)
+      index += 1
+    end
+
+    self
+  end
```

Comment:
- [ ] should have working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- Your method should return an enumerator when no block is given.
- It should work with ranges.
- It should work with hashes.